### ISSUE TITLE
Show in editor progress when connecting to a kernel

### ISSUE THREAD

**Issue by rebornix** [microsoft/vscode-jupyter#11161]
Jupyter extension presents all available Jupyter kernel specs, remote Jupyter server and Python environments to users when users open a notebook. The selected kernel doesn't necessarily have all the dependencies installed or might be slow to connect to, meaning when users click the run button on a cell, users are not likely to get execution results immediately. The delay might be seconds, or minutes if it requires downloading of ipykernel or Python extension, etc.

Currently, after users click the run button, we immediately show a pending icon to the cell status bar, indicating the cell is about to run once the extension/kernel is ready. The outline and run all button in the top toolbar will show a spinner too. These are the only two places where we show progress.

Jupyter extension might show a notification when it receives the request to start a kernel. For example, it shows a notification of the ipykernel installation progress for a new Python environment.

<img width="837" alt="image" src="https://user-images.githubusercontent.com/876920/185715172-509885af-8a6e-4c75-8fcf-2dd105087b3b.png">

From the image shown above, inside the notebook editor, there is no clear indicator that we are connecting to the kernel, except the pending icon on the cell status bar.

## Proposals

To make the connection state more intuitive, we can explore a few other UX affordances to help users understand what's going on after they click the run button

**1. Editor Progress Bar**

We can show a progress bar at the top of the notebook editor

<img width="957" alt="image" src="https://user-images.githubusercontent.com/876920/185714371-c4df1a7a-ad96-465d-a6a6-f3c91a107167.png">

**2. Spinner on Kernel Status Bar** 

We could turn the icon in the kernel status bar to a spinner, which can give a strong indication that the kernel is running

<img width="469" alt="image" src="https://user-images.githubusercontent.com/876920/185715548-922f5245-5009-49ec-a81f-c928a07c44f5.png">

**Comment by rebornix** [comment]
A good example of how we are missing progress in kernel connection is https://github.com/microsoft/vscode-jupyter/issues/9640#issuecomment-1220044454

**Comment by olguzzar** [comment]
I think this proposal would help an issue that I saw when running through the golden scenario. The kernel was still in the "connecting" state, but that was only apparent if you looked at the bottom status bar. While it was still connecting, I was then able to click "Run all", and "Go To" showed the spinning icon. As a user, this indicates to me that the notebook is actively running cells, when in reality, it's still trying to connect. Adding some sort of visual indicator that the kernel is connecting would be helpful. Additionally, it would be nice to have the action buttons, like Run All, disabled during the connecting phase.

Video recording of this scenario: https://msit.microsoftstream.com/video/79a60840-98dc-a27a-e9b5-f1ed287bbe75?st=149
